sdks/go/pkg/beam/core/runtime/exec/plan.go

```diff
@@ -87,6 +87,11 @@ func (p *Plan) ID() string {
 	return p.id
 }
 
+// ID returns the plan identifier.
+func (p *Plan) Source() *DataSource {
```

Comment:
Perhaps return just the necessary PtransformID, which has the bonus of documenting the intent? 
eg.
// SourcePTransformID returns the ID of the data's origin PTransform.
func SourcePTransformID() string {...}

Reply:
Good call. Done.